packages/discord.js/typings/index.d.ts

```diff
@@ -4733,14 +4738,14 @@ export type MessageChannelComponentCollectorOptions<T extends MessageComponentIn
 export interface MessageEditOptions {
   attachments?: MessageAttachment[];
   content?: string | null;
-  embeds?: (Embed | APIEmbed)[] | null;
+  embeds?: (JSONEncodable<APIEmbed> | APIEmbed)[] | null;
   files?: (FileOptions | BufferResolvable | Stream | MessageAttachment)[];
   flags?: BitFieldResolvable<MessageFlagsString, number>;
   allowedMentions?: MessageMentionOptions;
   components?: (
     | JSONEncodable<APIActionRowComponent<APIMessageActionRowComponent>>
-    | ActionRow<MessageActionRowComponent>
-    | (Required<BaseComponentData> & ActionRowData<MessageActionRowComponentData | MessageActionRowComponent>)
+    | ActionRowBuilder<MessageActionRowComponentBuilder>
+    | (Required<BaseComponentData> & ActionRowData<MessageActionRowComponentData | MessageActionRowComponentBuilder>)
```

Comment:
This is incompatible with the type of ActionRowData:
![image](https://user-images.githubusercontent.com/38259440/159136971-123a765d-bf71-429d-b587-ff68c084f774.png)

Reply:
The `ActionRowData` type seems to be incorrect. `ActionRowData` is used in `MessageOptions`, which is currently inaccurate because message options will either snake_case the json, or if it's a class it'll call `toJSON`. So if you pass `ActionRow` data with class components it wouldn't give the correct result. The only time classes are permitted in json is when passing it into an action row builder constructor.